controllers/clusterversion_controller.go

```diff
@@ -101,6 +172,49 @@ func (r *ClusterVersionReconciler) ensureConsolePlugin(clusterVersion string) er
 		return err
 	}
 
+	logger.Info("TEST TEST TEST TEST: Fetched the Deployment")
+
+	sub, err := GetOdfSubscription(r.Client)
+	if err != nil && !errors.IsNotFound(err) {
+		logger.Error(err, "failed to get ODF Subscription")
+		logger.Info(fmt.Sprintf("TEST TEST TEST TEST: Subscription err reason '%s'", errors.ReasonForError(err)))
+		logger.Info(fmt.Sprintf("TEST TEST TEST TEST: Subscription err code is 404 ?? '%t'", k8sUtil.IsKeyNotFoundError(err)))
+		return err
+	}
+
+	logger.Info("TEST TEST TEST TEST: Fetched the Subscription")
+	logger.Info(fmt.Sprintf("TEST TEST TEST TEST: Subscription name '%s'", sub.Name))
+	logger.Info(fmt.Sprintf("TEST TEST TEST TEST: Subscription namespace '%s'", sub.Namespace))
+	logger.Info(fmt.Sprintf("TEST TEST TEST TEST: Subscription deletionTimeStamp '%s'", sub.ObjectMeta.DeletionTimestamp))
+
+	// Deleting cluster-scoped resources which cannot be garbage collected,
+	// when either Subscription is marked for deletion or already deleted during this block's execution
+	if errors.IsNotFound(err) || k8sUtil.IsKeyNotFoundError(err) || !sub.GetDeletionTimestamp().IsZero() {
+
+		logger.Info("TEST TEST TEST TEST: Inside deletion condition")
+		logger.Info(fmt.Sprintf("TEST TEST TEST TEST: errors.IsNotFound(err) ?? '%t'", errors.IsNotFound(err)))
+		logger.Info(fmt.Sprintf("TEST TEST TEST TEST: k8sUtil.IsKeyNotFoundError(err) ?? '%t'", k8sUtil.IsKeyNotFoundError(err)))
+
+		odfConsolePlugin := console.GetConsolePluginCR(r.ConsolePort, OperatorNamespace)
+		err := r.Client.Delete(context.TODO(), odfConsolePlugin)
+		if err != nil && !errors.IsNotFound(err) {
+			logger.Error(err, "Failed to delete ConsolePlugin", "Name", odfConsolePlugin.ObjectMeta.Name)
+			return err
+		}
+		logger.Info("ConsolePlugin deleted successfully", "Name", odfConsolePlugin.ObjectMeta.Name)
+
+		consoleCLIDownload := console.GetConsoleCLIDownloadCR()
+		err = r.Client.Delete(context.TODO(), consoleCLIDownload)
+		if err != nil && !errors.IsNotFound(err) {
+			logger.Error(err, "Failed to delete ConsoleCLIDownload", "Name", consoleCLIDownload.ObjectMeta.Name)
+			return err
+		}
+		logger.Info("ConsoleCLIDownload deleted successfully", "Name", consoleCLIDownload.ObjectMeta.Name)
+
+		// return early if deletion successful
+		return nil
+	}
```

Comment:
this won't work as part of operator's uninstallation clean-up:
1. OLM doesn't have a way of knowing that there's some process the operator wants to complete before deleting the manager pod.
2. operator conditions api won't work, as they are for making sure OLM doesn't upgrade from one version of your operator to another before it finishes something.
3. There's no way for OLM to tell the controller the reason that it is being deleted. One reason might be that it is being uninstalled. Another reason might be that a node is being drained and the expectation is that the replica set will spin up another copy on another node.
4. The RBAC your controller needs in order to delete those objects might disappear before controller even executing.